### DISCUSSION TITLE
Running on managed server without Node or root access.

### DISCUSSION THREAD

**Discussion by benjibee** [spatie/browsershot#595] [Q&A] [answered]
We have a few websites running on the wonderful managed hosting provided by Kinsta. They offer quite a lot and we're really happy with them but Node isn't officially supported and my various attempts to get it up and running haven't been very successful. I have SSH access to the server which is running Ubuntu 20.04.3 LTS but no root access.

My question is: any way to get browsershot / Pupeteer up and running in such an environment?

Any ideas / sugestions would be great as we have many uses for this great package. Thanks!

**Comment by freekmurze** [comment, marked as answer]
Try running it via this cool package: https://github.com/stefanzweifel/sidecar-browsershot

**Reply by benjibee** [reply]
Oh yeah, this could really be something, how cool! I'll see if I can get it setup and report back here if there's any specifics for a non-Lambda project. Thanks @freekmurze!

**Reply by stefanzweifel** [reply]
@benjibee There's currently no stable version of that package yet. If you want to include it in your project, you have to install the `dev-main`-version of it.

```shell
composer require "wnx/sidecar-browsershot:dev-main"
```

A stable version will be tagged over the coming days.

**Reply by benjibee** [reply]
Thanks @stefanzweifel. I've just figured that part out a moment ago! The documentation confused me for a moment but I know this is a fairly new package.

Now I'm just seeing what I need to get it running without Lambda or Laravel, if you have any tips there. Though I don't mind tinkering around, either :)

**Reply by stefanzweifel** [reply]
@benjibee Yeah, the docs are a bit rought at the moment. The package got soft released only an hour ago. 😅 
Unfortunately, the package relies on [hammerstone/sidecar](https://github.com/hammerstonedev/sidecar) which itself requires Laravel. If you don't use Laravel, you're out of luck. 

Another option for you could be running Google Chrome on a different server by using [`setRemoteInstance`](https://github.com/spatie/browsershot#connection-to-a-remote-chromiumchrome-instance).

**Reply by benjibee** [reply]
I spent the day in the office trying to get Pupeteer to run on our host's servers. Eventually I gave up and started reliving old trauma seeing if dompdf somehow magically supports modern CSS (and PHP). But I got home from work and saw yours and Freek's tweets and hopped back on my laptop to try it out! So sorry for the undue pressure for your new project. I'll see what I can figure out with what I've got here. Thanks for all the help to both of you!
